sycl/plugins/opencl/CMakeLists.txt

```diff
@@ -9,9 +9,16 @@
 #TODO: remove dependency on pi.hpp in sycl project.
 #TODO: Currently, the pi.hpp header is common between sycl and plugin library sources. 
 #This can be changed by copying the pi.hpp file in the plugins project.
+configure_file(
+  ${CMAKE_CURRENT_SOURCE_DIR}/../../source/version.rc.in
+  ${CMAKE_CURRENT_BINARY_DIR}/version.rc
+  @ONLY)
+
+set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
```

Comment:
Isn’t it going to export all symbols on windows? Sounds opposite to what this patch is doing on Linux.

Reply:
> Isn’t it going to export all symbols on windows? Sounds opposite to what this patch is doing on Linux.

The name is confusing. I don't think this flag is used to export all symbols used in the library.
Here is what manual says about this flag.

> WINDOWS_EXPORT_ALL_SYMBOLS
> This property is implemented only for MS-compatible tools on Windows.
> 
> Enable this boolean property to automatically create a module definition (.def) file with all global symbols found in the input .obj files for a SHARED library on Windows. The module definition file will be passed to the linker causing all symbols to be exported from the .dll. For global data symbols, __declspec(dllimport) must still be used when compiling against the code in the .dll. All other function symbols will be automatically exported and imported by callers. This simplifies porting projects to Windows by reducing the need for explicit dllexport markup, even in C++ classes.
> 
> This property is initialized by the value of the CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS variable if it is set when a target is created.

Reply:
> I don't think this flag is used to export all symbols used in the library.

I understand the docs differently:
> All other function symbols will be automatically exported and imported by callers. This simplifies porting projects to Windows by reducing the need for explicit dllexport markup, even in C++ classes.

It says that *all* function symbols will be exported, and that is not what is done on Linux.

Reply:
@alexbatashev , Yes all FUNCTION symbols are exported. That's not ALL symbols.
Here is the proof that we only export pi* symbols on Windows:

```
bash-3.2$ dumpbin.exe /EXPORTS pi_opencl.lib
     Exports
                  piContextCreate
                  piDevicesGet
                  piEnqueueMemBufferMap
                  piEventCreate
                  piKernelCreate
                  piKernelSetExecInfo
                  piMemBufferCreate
                  piMemBufferPartition
                  piMemImageCreate
                  piPlatformsGet
                  piPluginInit
                  piProgramCreate
                  piProgramLink
                  piQueueCreate
                  piSamplerCreate
                  piclProgramCreateWithBinary
                  piclProgramCreateWithSource
                  piextContextCreateWithNativeHandle
                  piextDeviceCreateWithNativeHandle
                  piextDeviceSelectBinary
                  piextEventCreateWithNativeHandle
                  piextGetDeviceFunctionPointer
                  piextGetNativeHandle
                  piextKernelSetArgMemObj
                  piextKernelSetArgPointer
                  piextMemCreateWithNativeHandle
                  piextProgramCreateWithNativeHandle
                  piextProgramSetSpecializationConstantImpl
                  piextQueueCreateWithNativeHandle
                  piextUSMDeviceAlloc
                  piextUSMEnqueueMemAdvise
                  piextUSMEnqueueMemcpy
                  piextUSMEnqueueMemset
                  piextUSMEnqueuePrefetch
                  piextUSMFree
                  piextUSMGetMemAllocInfo
                  piextUSMHostAlloc
                  piextUSMSharedAlloc

bash-3.2$ dumpbin.exe /ALL pi_opencl.lib
    79 public symbols
     12D2 __IMPORT_DESCRIPTOR_pi_opencl
     1504 __NULL_IMPORT_DESCRIPTOR
     163Cpi_opencl_NULL_THUNK_DATA
     1BF6 __imp_piPluginInit
     1BF6 piPluginInit
     1B88 __imp_piPlatformsGet
     1B88 piPlatformsGet
     1808 __imp_piDevicesGet
     1808 piDevicesGet
     1F8C __imp_piextDeviceCreateWithNativeHandle
     1F8C piextDeviceCreateWithNativeHandle
     200C __imp_piextDeviceSelectBinary
     200C piextDeviceSelectBinary
     2102 __imp_piextGetDeviceFunctionPointer
     2102 piextGetDeviceFunctionPointer
     179A __imp_piContextCreate
     179A piContextCreate
     1F0A __imp_piextContextCreateWithNativeHandle
     1F0A piextContextCreateWithNativeHandle
     1D3C __imp_piQueueCreate
     1D3C piQueueCreate
     2468 __imp_piextQueueCreateWithNativeHandle
     2468 piextQueueCreateWithNativeHandle
     1A34 __imp_piMemBufferCreate
     1A34 piMemBufferCreate
     1B18 __imp_piMemImageCreate
     1B18 piMemImageCreate
     1AA4 __imp_piMemBufferPartition
     1AA4 piMemBufferPartition
     22E0 __imp_piextMemCreateWithNativeHandle
     22E0 piextMemCreateWithNativeHandle
     1C62 __imp_piProgramCreate
     1C62 piProgramCreate
     1E90 __imp_piclProgramCreateWithSource
     1E90 piclProgramCreateWithSource
     1E16 __imp_piclProgramCreateWithBinary
     1E16 piclProgramCreateWithBinary
     1CD0 __imp_piProgramLink
     1CD0 piProgramLink
     235E __imp_piextProgramCreateWithNativeHandle
     235E piextProgramCreateWithNativeHandle
     1954 __imp_piKernelCreate
     1954 piKernelCreate
     2268 __imp_piextKernelSetArgPointer
     2268 piextKernelSetArgPointer
     19C2 __imp_piKernelSetExecInfo
     19C2 piKernelSetExecInfo
     18E8 __imp_piEventCreate
     18E8 piEventCreate
     2082 __imp_piextEventCreateWithNativeHandle
     2082 piextEventCreateWithNativeHandle
     1DA8 __imp_piSamplerCreate
     1DA8 piSamplerCreate
     1874 __imp_piEnqueueMemBufferMap
     1874 piEnqueueMemBufferMap
     21F2 __imp_piextKernelSetArgMemObj
     21F2 piextKernelSetArgMemObj
     2812 __imp_piextUSMHostAlloc
     2812 piextUSMHostAlloc
     24E8 __imp_piextUSMDeviceAlloc
     24E8 piextUSMDeviceAlloc
     2882 __imp_piextUSMSharedAlloc
     2882 piextUSMSharedAlloc
     2730 __imp_piextUSMFree
     2730 piextUSMFree
     2646 __imp_piextUSMEnqueueMemset
     2646 piextUSMEnqueueMemset
     25D2 __imp_piextUSMEnqueueMemcpy
     25D2 piextUSMEnqueueMemcpy
     26BA __imp_piextUSMEnqueuePrefetch
     26BA piextUSMEnqueuePrefetch
     255A __imp_piextUSMEnqueueMemAdvise
     255A piextUSMEnqueueMemAdvise
     279C __imp_piextUSMGetMemAllocInfo
     279C piextUSMGetMemAllocInfo
     23E0 __imp_piextProgramSetSpecializationConstantImpl
     23E0 piextProgramSetSpecializationConstantImpl
     217E __imp_piextGetNativeHandle
     217E piextGetNativeHandle
```